opm/simulators/linalg/WellOperators.hpp

```diff
@@ -84,14 +84,43 @@ class WellModelAsLinearOperator : public LinearOperatorExtra<X, Y>
     void apply(const X& x, Y& y) const override
     {
         OPM_TIMEBLOCK(apply);
-        wellMod_.apply(x, y);
+        for (const auto& well : this->wellMod_) {
+            // Well equations B and C uses only the perforated cells, so need to apply on local vectors
```

Comment:
I think we can reduce duplication in a good way by moving the content of this loop to a separate function `applySingleWell()`, then the apply() function of both classes here would not duplicate each other.

Reply:
good idea. i have added a commit that does this. furthermore, applyscaleadd is also identical so i decided to share them.